TypeSystem/flow-analysis/reachability_A07_t07.dart

```diff
@@ -2,29 +2,23 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion If N is an expression of the form E1.m1(E2), then:
+/// @assertion Method invocation: If `N` is an expression of the form
+/// `E1.m1(E2)`, then:
+///  - Let `before(E1) = before(N)`
+///  - Let `before(E2) = after(E1)`
+///  - Let `T` be the static return type of the invocation
+///  - If `T <: Never` then:
+///   - Let `after(N) = unreachable(after(E2))`.
+///  - Otherwise:
+///    - Let `after(N) = after(E2)`
 ///
-///  Let before(E1) = before(N)
-///  Let before(E2) = after(E2)
-///  Let T be the static return type of the invocation
-///  If T <: Never then:
-///    Let null(N) = unreachable(before(N)).
-///    Let notNull(N) = unreachable(before(N)).
-///  Otherwise if T <: Null then:
-///    Let null(N) = before(N).
-///    Let notNull(N) = unreachable(before(N)).
-///  Otherwise if T is non-nullable then:
-///    Let null(N) = before(N).
-///    Let notNull(N) = unreachable(before(N)).
-///  Otherwise:
-///    Let null(N) = promote(x, Null, before(N))
-///    Let notNull(N) = promoteToNonNull(x, before(N))
-///
-/// @description Checks reachability after method call. Test non-nullable return
-/// type
+/// @description Checks reachability after method invocation. Tests a method
+/// with a non-nullable return type.
 /// @author sgrekhov@unipro.ru
 /// @issue 41985
 
+// @dart = 3.8
```

Comment:
Just to make sure: Dart 3.8 does _not_ have flow analysis that relies on sound null safety, which means that there's no need to update the comment below in line 32-33. I think it would be nice to have a comment to this effect.

Reply:
Updated here and below

Reply:
Thanks!